### DISCUSSION TITLE
Program not detecting ansi.sys

### DISCUSSION THREAD

**Discussion by nelgin** [dosemu2/dosemu2#1712] [General]
This is likely a bug but I'd like to double check my setup before opening it as a regression.

I have configured my userhook.sys to load nansi.sys just as I do in my config.sys in dosemu1.

When the door is run, it's printing 

[!This door requires you to be ANSI capable.Please load ANSI.SYS in your CONFIG.SYS and call back.
(the first character is a left arrow).
It happens when running both from the BBS and via the command line.
![image](https://user-images.githubusercontent.com/6232589/176724259-ca99638f-c48a-46c7-b283-0b2984de90d3.png)

![image](https://user-images.githubusercontent.com/6232589/176728881-6d884899-86f6-4ccb-8ee1-8405f908ede3.png)
/usr/bin/env  /usr/bin/dosemu.bin -q  -f/sbbs/ctrl/dosemu.conf -d /sbbs/node1/ -d /sbbs/xtrn -d /sbbs/ctrl/ -d /sbbs/data/ -d /sbbs/exec/

I have tried dosemu.bin with and without the -t option.

I can see that nansi.sys is loaded
6/30 16:20:54 term Node 1 <Nelgin> dmdtrv.exe: HIMEM is testing extended memory...
6/30 16:20:54 term Node 1 <Nelgin> dmdtrv.exe: done.
6/30 16:20:54 term Node 1 <Nelgin> dmdtrv.exe: NANSI.sys 4.0b: New ANSI driver.  Copyright 1986-2003 Daniel Kegel

Again, this works fine with dosemu1 so I'm adding the debug output to see if anyone can spot an issue with my setup.

[dosemu.conf.txt](https://github.com/dosemu2/dosemu2/files/9021706/dosemu.conf.txt)
[dosemu_boot.log.gz](https://github.com/dosemu2/dosemu2/files/9021708/dosemu_boot.log.gz)

If this appears to be a genuine regression, feel free to move to Issues.

**Comment by stsp** [comment]
When creating an issue,
don't forget to attach a test-case.

**Reply by nelgin** [reply]
[dmdtrv43.zip](https://github.com/dosemu2/dosemu2/files/9023688/dmdtrv43.zip)

Here's the zipfile that contains dmdtrv.exe. It'll run without a FOSSIL driver unless you tell it to (using /FD)

I'll also attach my nansi.sys which is a dropin replacement for ansi.sys - this works fine on dosemu1. (Obviously rename to remove the .txt)

[nansi.sys.txt](https://github.com/dosemu2/dosemu2/files/9023695/nansi.sys.txt)

**Reply by stsp** [reply]
Well, it will help if you stop using
custom configs, long command
lines, dosemu.bin instead of
dosemu, etc - at least for the time
of debugging.
I edited dmdtriv.cfg to point to
door.sys from YT, and it runs
without any error, both when
nansi is loaded or even not.

And now who knows if there is
something to do in that prog to
make it check for ansi, or maybe
some of your customizations
lead to problems?

**Comment by nelgin** [comment]
Well, you tell me exactly what you want me to try and I'll try it.

**Reply by stsp** [reply]
What I want it simple.
Run `dosemu`.
Then in DOS navigate to your
dmdtrv and run this:
`dmdtrv.exe dmdtriv.cfg`
If it says "can't read system file",
then in dmdtriv.cfg adjust the
path to door.sys and try again.
Its that simple.

**Comment by nelgin** [comment]
Don't worry about it. I'm done.
